server/test/mocks/mock_data_store.cpp

```diff
@@ -0,0 +1,58 @@
+#include "mock_data_store.hpp"
+
+namespace ins_service
+{
+
+::testing::NiceMock<MockDataStore>* g_mocked_data_store_;
+
+void DataStore::Init(const std::string& filename)
+{
+    EXPECT_TRUE(g_mocked_data_store_ != nullptr);
```

Comment:
Interesting use of `EXPECT_TRUE` inside the mock! I would have thrown an exception instead of something.